src/psyclone/psyad/domain/lfric/lfric_adjoint_harness.py

```diff
@@ -401,15 +500,25 @@ def generate_lfric_adjoint_harness(tl_psyir):
 
     # Finally, add the kernel itself to the list for the invoke().
     arg_nodes = []
-    for arg in kern_args.arglist:
-        arg_nodes.append(Reference(table.lookup(arg)))
+    for idx, arg in enumerate(kern_args.arglist):
+        sym = table.lookup(arg)
+        # Check whether this argument contains geometric information that
+        # is passed through from the Algorithm layer.
+        mdata_idx = kern_args.metadata_index_from_actual_index(idx)
+        if mdata_idx == coord_arg_idx:
+            sym = table.lookup_with_tag("coord_field")
```

Comment:
Might `lookup.arg()` get a different symbol to `lookup_with_tag("coord_field")`? It also doesn't feel right to lookup the symbol twice. Perhaps you could use continue and then move `table.lookup()` to the end of the loop?

Reply:
Yes, in general it will get a different symbol. However, from your questions I gather that my code isn't as readable as I thought it was :-( so I've added comments and moved this lookup to an `else` clause which probably makes more sense.